### ISSUE TITLE
Training output and settings explaination

### ISSUE THREAD

**Issue by simoneangarano** [facebookresearch/DomainBed#53]
Hi!
Thank you for your repository. I'd like to have some clarification about the training output; setting for example test_envs to 0 I get this output:
![image](https://user-images.githubusercontent.com/74110683/119803055-6536b280-bedf-11eb-905f-5cb2a9446991.png)
What does the columns represent?
Moreover, what's the difference between different test_envs configurations?
Thank you for your help!

**Comment by daysm** [comment]
On which dataset are you evaluating? In the case of Office-Home for example, the environments `env0` to `env3` would respond to the domains `A`, `C`, `P`, and `R` respectively.

From section 5 in the paper:

> We split the data from each domain into 80% and 20% splits. We use the larger splits for training and final evaluation, and the smaller splits to select hyperparameters (for an illustration, see Appendix B.3)

`env_x_in` is the larger split, `env_x_out` is the smaller split.

This is the illustration from Appendix B.3:

![Screenshot 2021-05-27 at 12 05 24](https://user-images.githubusercontent.com/7551272/119807846-d5dfce00-bee3-11eb-99e1-803b9d30ad8b.png)

**Comment by simoneangarano** [comment]
Ok thanks for your reply! So, the results on the testing domains are only given at the end of the whole procedure?
Moreover, what about the test_envs parameter?

**Comment by daysm** [comment]
If you pass `--test_envs 0` to the training script, then that means you train on the remaining envs `1`, `2`, and `3` and use `0` only for testing. If you were to pass `--test_envs 0 1` to the training script, you would only train on `2` and `3` and use both `0` and `1` only for testing.

At each training checkpoint (in your case after every step, for PACS the default is every 300 steps) you evaluate the model on all splits of the data. This is necessary later for model selection.

For the model selection method Leave-one-domain-out validation, you need two test domains: you evaluate on one domain to find the best model, but then you report the accuracy on the other test domain.

The relation between dataset split, test environments and model selection is explained in Appendix B.3.

**Comment by lopezpaz** [comment]
Thank you @daysm

**Comment by GA-17a** [comment]
@daysm Hi!
After finding the best model using a model selection method, do you report the results of the last checkpoint in your paper?
